tensorflow_io/core/python/experimental/avro_record_dataset_ops.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import tensorflow as tf
+from tensorflow.python.data.ops.readers import (
+    _create_or_validate_filenames_dataset,
+    _create_dataset_reader,
+)
+from tensorflow.python.data.util import convert
+from tensorflow_io.core.python.ops import core_ops
+
+_DEFAULT_READER_BUFFER_SIZE_BYTES = 256 * 1024  # 256 KB
+_DEFAULT_READER_SCHEMA = ""
+# From https://github.com/tensorflow/tensorflow/blob/v2.0.0/tensorflow/python/data/ops/readers.py
+
+
+class _AvroRecordDataset(tf.data.Dataset):
+    """A `Dataset` comprising records from one or more AvroRecord files."""
+
+    def __init__(self, filenames, buffer_size=None, reader_schema=None):
+        """Creates a `AvroRecordDataset`.
+        Args:
+          filenames: A `tf.string` tensor containing one or more filenames.
+          buffer_size: (Optional.) A `tf.int64` scalar representing the number of
+            bytes in the read buffer. 0 means no buffering.
+          reader_schema: (Optional.) A `tf.string` scalar representing the reader schema or None
+        """
+        self._filenames = filenames
+        self._buffer_size = convert.optional_param_to_tensor(
```

Comment:
The convert.optional_param_to_tensor is pretty simple:

https://github.com/tensorflow/tensorflow/blob/f40a875355557483aeae60ffcf757fc9626c752b/tensorflow/python/data/util/convert.py#L26-L35

Can you just copy the content here? The following should work:
```
def optional_param_to_tensor(argument_name,
                             argument_value,
                             argument_default=0,
                             argument_dtype=dtypes.int64):
  if argument_value is not None:
    return tf.convert_to_tensor(
        argument_value, dtype=argument_dtype, name=argument_name)
  else:
    return tf.constant(
        argument_default, dtype=argument_dtype, name=argument_name)

```

This will avoid future changes in TF that could break tensorflow/io

Reply:
thanks, updated.